OpenRA.Mods.D2k/Traits/Sandworm.cs

```diff
@@ -71,7 +71,12 @@ public override void OnBecomingIdle(Actor self)
 			if (withSpriteBody.DefaultAnimation.CurrentSequence.Name != Info.IdleSequence)
 				withSpriteBody.DefaultAnimation.PlayRepeating(Info.IdleSequence);
 
-			base.OnBecomingIdle(self);
+			var thumper = self.World.FindActorsInCircle(self.CenterPosition, Info.MaxSearchRadius)
```

Comment:
This is also easy to do: add `&& attackTrait.HasAnyValidWeapons(Target.FromActor(x))` to the `FirstOrDefault` in the target check, and then give it a target type that the worm isn't valid against.